### PR TITLE
Domains Management i1: Initial domains table header with sorting

### PR DESCRIPTION
**Pull request by vykes-mac**
Related to https://github.com/Automattic/dotcom-forge/issues/3423

## Proposed Changes
This PR setup initial sorting for the domains table by:
* Adding `domains-table-header` component in the domains table package
* Add  `DomainsTableColumn` type that reflects the structure of the domain table and the header
* Include sorting function that's associated with each column/header

As it is the data is not currently sorted as sorting works on the `DomainData` which is lazy loaded as rows come into view. This PR doesn't handle that, I suspect we will need to eventually keep track of the `DomainData` data when it's loaded in the parent of the DomainsTableRow so that we can sort it and re-render the rows with the sorted data.

For this PR we simple add the `domains` header with the chevron indicating it's sortable.

## Testing Instructions

* browse `domains/manage?flags=domains/management`
* Ensure the header shows up with a single `domains` column
* Click the `Domains` header and the chevron should toggle to indicate the sortDirection. 

Next:

- [ ] Add tests

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/domains-table/src/domains-table-header/style.scss [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	display: flex;
+	justify-content: space-between;
+	padding: 8px 0 16px;
+	font-size: $font-body-small;
+	font-weight: 500;
+	color: var(--color-neutral-60);
+	border-bottom: 1px solid #f0f0f0;
+
+	.button-plain {
+		font-family: "SF Pro Text", $sans;
```
Comment (p-jackson) [inline packages/domains-table/src/domains-table-header/style.scss]: We don't need to set the font here, even though that's what they do in Figma. "SF Pro Text" is only available on Mac. But our global CSS styles already ensure that San Francisco is used on Mac.

File: client/my-sites/domains/domain-management/list/bulk-all-domains.tsx [hunk 2]
```diff
@@ -41,13 +42,24 @@ export default function BulkAllDomains( props: BulkAllDomainsProps ) {
...
+	];
+
 	return (
 		<>
 			<PageViewTracker path={ props.analyticsPath } title={ props.analyticsTitle } />
 			<Main wideLayout>
 				<BodySectionCssClass bodyClass={ [ 'edit__body-white' ] } />
 				<DomainHeader items={ [ item ] } buttons={ buttons } mobileButtons={ buttons } />
-				<DomainsTable domains={ domains } />
+				<DomainsTable domains={ domains } columns={ domainsTableColumns } />
```
Comment (p-jackson) [inline client/my-sites/domains/domain-management/list/bulk-all-domains.tsx]: I'm wondering whether we really need to accept `columns` as a prop. We want to be able to share the `<DomainsTable>` component, but I can't imagine we want each place to provide their own columns?

The other thing to keep in mind is that not every column in the table is going to correspond with a field in the `DomainData` object. There's going to be columns with buttons for resolving errors. And there's the column with the ellipsis button too.

This approach does allow us to share the existing `getSimpleSortFunctionBy` function which is good. I think this functionality should be part of the `domains-table` package though. Someone else using the `<DomainsTable>` in another context is going to need to reimplement all the sorting themselves.

If we do want to keep the columns customisable, maybe the way to do that would be to have the `useDomainsTable` hook return the columns data. That's how I imagined allowing the user to customise things. The logic for fetching domains data (or in this case setting up columns) is still handled in the package. But if the user of the component wants to filter or modify that stuff before passing it to the component then they can do that.

**Review by p-jackson — COMMENTED** [review]
I've left some comments below on what I think about having customisable columns.

Also could you try it out in Storybook? At the moment there's an error I think because the columns prop is required but it hasn't been added to storybook.

You can start storybook with `yarn storybook:start`

File: client/my-sites/domains/domain-management/list/bulk-all-domains.tsx (hunk 2 above)
Comment (vykes-mac) [reply client/my-sites/domains/domain-management/list/bulk-all-domains.tsx]: I see your point, It's probably best to encapsulate columns as part of the domains table and move the neccessary sorting utils to the package also. Maybe in the future there might be a need for customisable column but that seem like we should be changing the column in the domains-table package if that's the case.

Comment (zaguiini) [reply client/my-sites/domains/domain-management/list/bulk-all-domains.tsx]: >I'm wondering whether we really need to accept columns as a prop. We want to be able to share the <DomainsTable> component, but I can't imagine we want each place to provide their own columns?

I resonate with that. I think that now that we have the `isAllSitesView` prop we can switch the columns according to that prop.

>Maybe in the future there might be a need for customisable column but that seem like we should be changing the column in the domains-table package if that's the case.

Yes 👍

Comment (vykes-mac) [reply client/my-sites/domains/domain-management/list/bulk-all-domains.tsx]: > The other thing to keep in mind is that not every column in the table is going to correspond with a field in the DomainData object. There's going to be columns with buttons for resolving errors. And there's the column with the ellipsis button too.

I was thinking about adding a `renderCell` function in `DomainsTableColumn` so a custom component can be rendered in that column cell if we like.

Comment (zaguiini) [reply client/my-sites/domains/domain-management/list/bulk-all-domains.tsx]: >I was thinking about adding a renderCell function in DomainsTableColumn so a custom component can be rendered in that column cell if we like.

That can work, but we can revisit that once it's time to. Let's try and keep this as simple as possible 👌

**Review by p-jackson — APPROVED** [review]
I made a couple tweaks and was about to merge this until I noticed you still had a TODO to add some test.

The tweaks I made were:
- Switching from `translate()` to `__()`. Because we're in a package we need to be able to pass in the `domain` argument for translating, and that's a feature that comes with the `@wordpress/i18n` package. The domain is there because the app that imports the `@automattic/domains-table` package is ultimately responsible for supplying the translations, and the `__i18n_text_domain__` placeholder allows them to say what domain the translations come from.
- I changed the key React uses when iterating over column headings. We don't need to resort to the index because each `DomainsTableColumn` comes with it's own unique ID.

**Pull request merged**